circle.yml

```diff
@@ -4,18 +4,22 @@ machine:
   environment:
     _JAVA_OPTIONS: "-Xmx3G -Xms2G"
 dependencies:
-  pre: 
+  pre:
     - sudo apt-get update -y; true
     - sudo apt-get install -y apt gnupg2 gnupg-curl
     - sudo add-apt-repository ppa:ethereum/ethereum -y
     - sudo apt-get update -y; true
     - sudo apt-get install -y solc
     - gpg2 --keyserver https://secchannel.rsk.co/release.asc --recv-keys 5DECF4415E3B8FA4
+    - wget https://sonarsource.bintray.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-$SONAR_VERSION-linux.zip
+    - unzip sonar-scanner-cli-$SONAR_VERSION-linux.zip
   override:
     - gpg2 --verify SHA256SUMS.asc
     - sha256sum --check SHA256SUMS.asc
     - ./configure.sh
     - ./gradlew dependencies
 test:
   override:
-    - ./gradlew test
\ No newline at end of file
+    - ./gradlew shadow test
+  post:
+    - if [ ! -z $CI_PULL_REQUEST ]; then  export PULL_REQUEST=$(echo $CI_PULL_REQUEST | awk '{print substr($1,39)}') && ./sonar-scanner-$SONAR_VERSION-linux/bin/sonar-scanner -Dsonar.analysis.mode=preview -Dsonar.github.pullRequest=$PULL_REQUEST -Dsonar.github.repository=rsksmart/rskj -Dsonar.github.oauth=$GITHUB_ACCESS_TOKEN -Dsonar.host.url=$SONAR_HOST -Dsonar.login=$SONARQUBE_ACCESS_TOKEN -Dsonar.projectKey="co.rsk:rskJ" -Dsonar.sources="rskj-core/src/main/java,rskj-core/src/main/resources" -Dsonar.java.binaries="rskj-core/build/classes/java/main"; fi
```

Comment:
Next time we might want to extract a script for this if things start getting more complicated :)